app/lib/actions/katello/repository/clone_yum_metadata.rb

```diff
@@ -8,7 +8,7 @@ def plan(source_repo, target_repo)
             # actually updated, but it is not reflected in the database yet.
             output = {}
             if !target_repo.root.previous_changes.include?(:unprotected) &&
-                target_repo.environment
+                target_repo.environment && !target_repo.environment.library
```

Comment:
Can you explain (gap in my knowledge here...) how `!target_repo.environment.library` means "we are publishing to library"? To me that sounds like "if the target repository environment is not library"

Let me know if I haven't made sense :)

Reply:
So what we want to do is to not check matching content and hence not set the matching_content param here https://github.com/Katello/katello/blob/8162d767b895366c9b23f1c4a38e523f44b743d1/app/lib/actions/katello/repository/clone_yum_metadata.rb#L22

That would cause the MetadataGenerate action to actually generate metadata..Right now since sha is not being checked, it passes matching_content as true and the metadata generation is skipped on the library repo.

Reply:
Thank you that makes sense! The change seems reasonable. Would like to have @parthaa think it through as well before we merge :+1: